CMakeLists.txt

```diff
@@ -1811,12 +1811,20 @@ if(CONFIG_CHECK_INIT_PRIORITIES)
   list(APPEND
     post_build_commands
     COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/check_init_priorities.py
-      --build-dir ${PROJECT_BINARY_DIR}/..
-      --edt-pickle ${EDT_PICKLE}
+      --elf-file=${ZEPHYR_BINARY_DIR}/${KERNEL_ELF_NAME}
```

Comment:
Failing with `armclang` with following error:
```
Traceback (most recent call last):
  File "/projects/github/ncs/zephyr/scripts/build/check_init_priorities.py", line 394, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/projects/github/ncs/zephyr/scripts/build/check_init_priorities.py", line 376, in main
    validator = Validator(args.elf_file, args.edt_pickle, log)
  File "/projects/github/ncs/zephyr/scripts/build/check_init_priorities.py", line 253, in __init__
    self._obj = ZephyrInitLevels(elf_file_path)
  File "/projects/github/ncs/zephyr/scripts/build/check_init_priorities.py", line 110, in __init__
    self._load_level_addr()
  File "/projects/github/ncs/zephyr/scripts/build/check_init_priorities.py", line 145, in _load_level_addr
    raise ValueError(f"Missing init symbols, found: {self._init_level_addr}")
ValueError: Missing init symbols, found: {}
ninja: build stopped: subcommand failed.
```

Reply:
Bummer. Would you be able to send an elf file over to me?

Reply:
I'd honestly not block this PR because it doesn't work with armclang, but exclude the target if the toolchain being used is armclang. It can be investigated later by users with access to it. Restricting progress on OSS projects because of licensed software is not a good idea.

Reply:
does it work with public clang?

Reply:
```
$ export ZEPHYR_TOOLCHAIN_VARIANT=llvm
$ west build -p -b native_posix_64 samples/subsys/display/lvgl
...
-- Found toolchain: host (clang/ld)
...
-- The C compiler identification is Clang 14.0.6
-- The CXX compiler identification is Clang 14.0.6
-- The ASM compiler identification is Clang with GNU-like command-line
...
[329/329] Linking C executable zephyr/zephyr.elf
$ ./scripts/build/check_init_priorities.py -v
INFO: check_init_priorities: build/zephyr/zephyr.elf
INFO: /gpio@800/sdl_gpio POST_KERNEL 3 > /gpio@800 POST_KERNEL 2
INFO: /keys POST_KERNEL 5 > /gpio@800 POST_KERNEL 2
INFO: /keys POST_KERNEL 5 > /gpio@800 POST_KERNEL 2
INFO: /lvgl_button_input APPLICATION 2 > /keys POST_KERNEL 5
INFO: /lvgl_pointer APPLICATION 3 > /input-sdl-touch POST_KERNEL 6
$ ./scripts/build/check_init_priorities.py -i
EARLY
PRE_KERNEL_1
  __init_statics_init_pre: statics_init(NULL)
  __init___device_dts_ord_18: np_uart_0_init(__device_dts_ord_18)
  __init_uart_console_init: uart_console_init(NULL)
  __init_posix_arch_console_init: posix_arch_console_init(NULL)
PRE_KERNEL_2
  __init_sys_clock_driver_init: sys_clock_driver_init(NULL)
POST_KERNEL
  __init_enable_logger: enable_logger(NULL)
  __init_enable_shell_uart: enable_shell_uart(NULL)
  __init___device_dts_ord_31: gpio_emul_init(__device_dts_ord_31)
  __init___device_dts_ord_32: gpio_sdl_init(__device_dts_ord_32)
  __init_k_sys_work_q_init: k_sys_work_q_init(NULL)
  __init___device_dts_ord_10: gpio_keys_init(__device_dts_ord_10)
  __init___device_dts_ord_12: sdl_init(__device_dts_ord_12)
APPLICATION
  __init___device_dts_ord_16: sdl_display_init(__device_dts_ord_16)
  __init_lvgl_init: lvgl_init(NULL)
  __init___device_dts_ord_11: lvgl_button_input_init(__device_dts_ord_11)
  __init___device_dts_ord_13: lvgl_pointer_input_init(__device_dts_ord_13)
SMP
```

Yeah looks alright to me, the armglang linker must be discarding the symbols, maybe the the binary is stripped? Hard to say without a file...